wordpress-mobile/WordPress-iOS
# Jetpack: Reorganize My Site Menu

## Requirement 

Re-organize My Site menu items into 3 sections:

- **Content**, the bits of information published on a site.
- **Traffic**, the flow and study of visitors viewing a site.
- **Maintenance**, the ongoing things that help keep a site running smoothly.

Source pcdRpT-1Vl-p2

## Implementation Details

1. The `My Site > Menu` screen data is stored in an array called `tableSections` of type `NSArray<BlogDetailsSection>`. And each `BlogDetailsSection` object has a list of rows of type `NSArray<BlogDetailsRow>`.
2. The method setting the `tableSections` array is `configureTableViewData`. So this menu reorganization is done by refactoring the `configureTableViewData` method.
3. These changes are only applied to Jetpack. See [comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/20476/files#r1157857588).
4. Each section ( Content, Traffic, and Maintenance ) is a method in `BlogDetailsViewController`.
    - -(BlogDetailsSection *)contentSectionViewModel
    - -(BlogDetailsSection *)trafficSectionViewModel
    - -(BlogDetailsSection *)maintenanceSectionViewModel
5. The rows implementation was embedded in the existing `sectionViewModel` methods ( see `- (BlogDetailsSection *)jetpackSectionViewModel` method for example ). So to avoid code duplication, I extracted those rows into properties of `BlogDetailsViewController` and re-used them in the new `sectionViewModel` methods.

## Screenshots

| Before | After |
| ------- | ----- |
| ![](https://user-images.githubusercontent.com/9609223/229856757-7dded6da-15f3-4917-ae61-669656433bf9.png) | ![](https://user-images.githubusercontent.com/9609223/229856740-3b837db6-dbb1-47bf-a384-7020b635ec0b.png) |

## Test Instructions

I recommend testing this PR by comparing the new menu with the old one and ensuring the same rows are visible. To do that, either get 2 devices and build & run this PR in one of the devices or install the **Jetpack Alpha** build alongside the one from AppStore.

Please repeat the following steps using:
- a wordpress.com site with admin rights ( for example a personal site ).
- a wordpress.com site you don't own.
- a self-hosted site you own ( you can use jurassic.ninja ).

1. Make sure you have Jetpack from AppStore installed and logged in.
2. Build and run this PR and log in.
3. Head to **My Site > Menu** screen.
4. **Verify** the rows are grouped in **Content**, **Traffic** and **Maintenance** sections.

## Regression Notes
1. Potential unintended areas of impact
A potential regression bug could be a row visible in older versions and no longer visible in this version or rows that weren't visible before and are now visible. This PR should be tested with wp.com and self-hosted sites you own and other sites you don't own.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and comparing with the version in AppStore.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
